src/extra/widgets/animbtn/lv_animbtn.h

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file lv_animbtn.h
+ *
+ */
+
+#ifndef LV_ANIMBTN_H
+#define LV_ANIMBTN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../widgets/lv_img.h"
+
+#if LV_USE_ANIMBTN != 0
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef enum {
+    LV_ANIMBTN_STATE_RELEASED = 1,          /* 0001 */
+    LV_ANIMBTN_STATE_PRESSED  = 2,          /* 0010 */
+    LV_ANIMBTN_STATE_DISABLED = 3,          /* 0100 */
```

Comment:
`3 != /* 0100 */`

Reply:
The bit pattern here is used to track the transition via a graycode (that was my initial idea, to check if a transition applies with a XOR, instead of "if from == prev_state && to == new_state"). But in the end there's only few states so the complexity of gray coding for this optimization is too large for the expected gain. I'll remove the comment, since it doesn't fit anymore.